src/common/selftest/selftest_MINI.cpp

```diff
@@ -234,10 +247,30 @@ void CSelftest::log_open() {
         serial_otp = 0;
     }
     char serial[32] = "unknown";
-    char fname[32] = "test_unknown.txt";
+    const char *suffix = "";
+    switch (m_Mask) {
+    case stmFans:
+        suffix = "_fans";
```

Comment:
I recommend using `static const char fans[] = "_fans"; suffix = fans;`
This makes sure the compiler won't try to load the "_fans" onto stack first, but uses the address into CPU FLASH

Reply:
i do not see any benefits, all the strings constants are placed in the flash, nothing loaded onto stack.